sdks/python/apache_beam/transforms/core.py

```diff
@@ -220,15 +222,29 @@ def __init__(self, generator, *args, **kwargs):
     self._gen_args = args
     self._gen_kwargs = kwargs
 
+  def element_generator(self):
+    while self._current_value is not None:
+      yield self._current_value
+
+    raise StopIteration()
+
   def start_bundle(self, context):
-    self._generator = self._generator_def(*self._gen_args, **self._gen_kwargs)
-    self._generator.send(None)
+    self._generator = self._generator_def(
+        self.element_generator(), *self._gen_args, **self._gen_kwargs)
 
   def process(self, context, *args, **kwargs):
-    return self._generator.send((context, args, kwargs))
+    self._current_value = (context, args, kwargs)
```

Comment:
Arg. Actually I think there's still problems with fan-in, I see what you mean by needing to switch from push to pull. It's the same reason you can't support flattening the returned values of the iterator as you want above. The pipeline needs the 1:1 correspondence between consuming values and returned values, and this restriction seems to be enforced on the actual user code sent to the runner not artificially placed by the SDK. This seems like pretty steep restrictions to put on user code. and maybe I'm misunderstanding, but reading through the source, there doesn't seem to be a "lower level" in the SDK that I can hook into to avoid this restriction.